streams/src/main/java/org/apache/kafka/streams/processor/internals/SinkNode.java

```diff
@@ -97,6 +96,10 @@ public void process(final K key, final V value) {
                                     keyClass,
                                     valueClass),
                     e);
+        } catch (final StreamsException e) {
```

Comment:
Because this upper layer can provide a more meaningful exception with more context.